File: Loops_PetkoLyutskanov/src/CheckTheSum.java
import java.util.HashMap;

public class CheckTheSum {
    public static boolean meetsTheTarget(String firstWord, String secondWord, String targetWord) {
        HashMap<Character, Integer> charValuesMap = new HashMap<>();

        int value = 0;
        int firstWordSum = 0;
        int secondWordSum = 0;
        int targetWordSum = 0;

        for (char ch = 'a'; ch <= 'z'; ++ch) {
            charValuesMap.put(ch, value);
            value++;
        }

        for (int i = 0; i < firstWord.length(); i++) {
            firstWordSum += charValuesMap.get(firstWord.charAt(i));
            secondWordSum += charValuesMap.get(secondWord.charAt(i));
            targetWordSum += charValuesMap.get(targetWord.charAt(i));
        }

        if (firstWordSum == secondWordSum && firstWordSum + secondWordSum == targetWordSum) {
            return true;
        } else {
            return false;
        }
    }

    public static void main(String[] args) {
        String firstWord = "acb";
        String secondWord = "cba";
        String targetWord = "cdb";

        System.out.println(meetsTheTarget(firstWord, secondWord, targetWord));
    }
}

File: Loops_BozhidarGenchev/checkTheSum.js
function checkSum (firstWord,secondWord,targetWord){

const firstWordArr=firstWord.split('')

const firstWordSum=firstWordArr.reduce((a,x)=>{
    if(x!=='a'){
        a+=x.charCodeAt()-97
    }
   return a;
   
},0)
const secondWordArr=secondWord.split('')

const secondWordSum=secondWordArr.reduce((a,x)=>{
    if(x!=='a'){
        a+=x.charCodeAt()-97
    }
   return a;
},0)

const tergetWordArr=targetWord.split('')

const targetWordSum=tergetWordArr.reduce((a,x)=>{
    if(x!=='a'){
        a+=x.charCodeAt()-97
    }
   return a;
},0)



return firstWordSum+secondWordSum===targetWordSum;

}
console.log(checkSum("acb","cba","cdb"))

//newSolutionWithArrowFunc
const checkSum = (firstWord, secondWord, targetWord) => {
  const charCodeSum = (str) => {
    let sum = 0;
    for (let i = 0; i < str.length; i++) {
      if (str[i] !== 'a') {
        sum += str[i].charCodeAt(0) - 97;
      }
    }
    return sum;
  }
  return charCodeSum(firstWord) + charCodeSum(secondWord) === charCodeSum(targetWord);
}
console.log(checkSum("acb", "cba", "cdb"));
